### PR TITLE
Start wiring up fragment program for OpenGLES

### PR DESCRIPTION
**Pull request by dnfield**
Makes fragment programs work for OpenGLES backend.

Fixes https://github.com/flutter/flutter/issues/113715
Fixes https://github.com/flutter/flutter/issues/105538
I cannot find a dedicated issue for this, but there probably is someone somewhere and I don't want to file a new one for it.

### CODE REVIEW

File: impeller/aiks/aiks_unittests.cc [hunk 1]
```diff
@@ -2943,13 +2943,15 @@ TEST_P(AiksTest, CanRenderForegroundAdvancedBlendWithMaskBlur) {
 
 // Regression test for https://github.com/flutter/flutter/issues/126701 .
 TEST_P(AiksTest, CanRenderClippedRuntimeEffects) {
-  if (GetParam() != PlaygroundBackend::kMetal) {
+  if (GetParam() == PlaygroundBackend::kVulkan) {
```
Comment (chinmaygarde) [inline impeller/aiks/aiks_unittests.cc:2946]: Perhaps move this to `PlaygroundTest::BackendSupportsFragmentProgram`?

File: impeller/renderer/backend/gles/buffer_bindings_gles.cc [hunk 2]
```diff
@@ -312,6 +314,9 @@ bool BufferBindingsGLES::BindUniformBuffer(const ProcTableGLES& gl,
             );
             continue;
         }
+        VALIDATION_LOG << "Size " << member.size
```
Comment (chinmaygarde) [inline impeller/renderer/backend/gles/buffer_bindings_gles.cc:317]: Can't this information be combined in the validation log below?

File: impeller/playground/playground.h [hunk 3]
```diff
@@ -30,6 +31,18 @@ enum class PlaygroundBackend {
   kVulkan,
 };
 
+inline RuntimeStageBackend PlaygroundBackendToRuntimeStageBackend(
```
Comment (chinmaygarde) [inline impeller/playground/playground.h:34]: `constexpr` with an FML_UNREACHABLE at the end.

**Review by chinmaygarde — APPROVED** [review]

**Comment by chinmaygarde** [comment]
Perhaps https://github.com/flutter/flutter/issues/105538 as a related issue.

File: impeller/renderer/backend/gles/buffer_bindings_gles.cc (hunk 2 above)
Comment (dnfield) [reply impeller/renderer/backend/gles/buffer_bindings_gles.cc:317]: This one is specific for floats.  The one below doesn't necessarily care about sizes, and when I hit this it was a little confusing at first about why a float wouldn't be supported.

**Review by bdero — APPROVED** [review]

File: impeller/playground/playground.h (hunk 3 above)
Comment (dnfield) [reply impeller/playground/playground.h:34]: Done

File: impeller/aiks/aiks_unittests.cc (hunk 1 above)
Comment (dnfield) [reply impeller/aiks/aiks_unittests.cc:2946]: Done

**Pull request merged**